zsign.cpp

```diff
@@ -64,6 +65,31 @@ int usage()
 	return -1;
 }
 
+string getCacheDirectory() {
+    try {
+#ifdef _WIN32
+        const char* buffer = std::GetEnvironmentVariable("LOCALAPPDATA", nullptr, 0);
```

Comment:
Please, see documentation (https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getenvironmentvariable) for `GetEnvironmentVariable()`:
- Second argument should be pointer to a sufficiently-large buffer.
- Third argument should be size of that buffer.
- Return value is number of characters copied.

Also, `Winbase.h` should be `#include`'ed; and I don't think function will appear in the `std` namespace (but in the global scope instead).